apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -146,9 +151,23 @@ def timestamp(self) -> Optional[datetime.datetime]:
             return None
         return ms_to_datetime(self.timestamp_ms)
 
+    @property
+    def native_context(self) -> clib.SOMAContext:
+        """The C++ SOMAContext for this SOMA context."""
+        with self._lock:
+            if self._native_context is not None:
+                return self._native_context
+
+        # tiledb_config also acquires the lock so we must first release it and
+        # then reacquire it
+        cfg = self.tiledb_config
```

Comment:
I think you just want to call `self._internal_tiledb_config()` here while holding onto the lock. Right now this ends up creating a race condition where we might end up constructing two separate `clib.SOMAContext`s and using them at different times.